spec/parser/magic_spec.cr

```diff
@@ -0,0 +1,56 @@
+require "../spec_helper"
+
+describe LC::Parser do
+  context "magic vars" do
+    it "parses magic line expressions" do
+      node = parse_expr <<-EXPR
+
+
+      a = __LINE__
+      EXPR
+
+      node.should be_a LC::Assign
+      node = node.as(LC::Assign)
+
+      node.value.should be_a LC::IntLiteral
+      node.value.as(LC::IntLiteral).value.should eq 2
```

Comment:
Shouldn't this be 3? I don't believe the line count starts from 0.

Reply:
The expr being parsed is `"\n\na = __LINE__"`. IIRC heredocs line 0 is the first line after the opener

Reply:
Right, but the heredoc isn't part of the parsed content: https://carc.in/#/r/h150

Reply:
Ahh I understand now, my bad.